ivy/func_wrapper.py

```diff
@@ -660,6 +665,34 @@ def _handle_view_indexing(*args, **kwargs):
     return _handle_view_indexing
 
 
+def _convert_numpy_arrays_to_torch(*args):
+    torch_tensors = []
+    for array in args:
+        if isinstance(array, np.ndarray):
```

Comment:
Ideally, we should do a nested search as the argument may be a list containing numpy arrays, which wouldn't work in that case. maybe make use of ivy nest functions here.

Reply:
Adding to this, it would be great if we could try and reuse/refactor some of the logic [here](https://github.com/unifyai/ivy/blob/d5cc0291e64af78f58daef7650d9f75a6625df50/ivy/utils/backend/handler.py#L366) at it serves a similar purpose (except for the garbage collection part). Thanks!

Reply:
I think what @CatB1t meant here is that we do something like,
```
np_arr_idxs = ivy.nested_argwhere(args)        # nested search
np_arr_vals = ivy.multi_index_nest(args)         # indexing to get the arrays  
backend_arr_vals = [ivy.asarray(val).to_native() for val in np_arr_vals]      # conversion
ivy.set_nest_at_indices(args, np_arr_idxs, backend_arr_vals)   # setting the backend ones back
```